packages/account/src/Sections/Security/LoginHistory/login-history.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Loading, ThemedScrollbars } from '@deriv/components';
+import { observer, useStore } from '@deriv/stores';
+import { WS } from '@deriv/shared';
+import LoadErrorMessage from 'Components/load-error-message';
+import LoginHistoryContent from './login-history-content';
+import { getLoginHistoryFormattedData } from '../../../../../utils/src/getLoginHistoryFormattedData';
+
+type TData = { id: number; date: string; action: string; browser: string; ip: string; status: string }[];
+
+const LoginHistory = observer(() => {
+    const { client, ui } = useStore();
+    const { is_switching } = client;
+    const { is_mobile } = ui;
+    const [is_loading, setLoading] = React.useState(true);
+    const [error, setError] = React.useState('');
+    const [data, setData] = React.useState<TData>([]);
+
+    React.useEffect(() => {
+        let is_cancelled = false;
+        const fetchData = async () => {
+            if (is_cancelled) return;
+
+            const api_res = await WS.authorized.fetchLoginHistory(50);
+            setLoading(false);
+            if (api_res.error) {
+                setError(api_res.error.message);
+            } else {
+                const formatted_data = getLoginHistoryFormattedData(api_res.login_history);
+                setData(formatted_data);
+            }
+        };
+
+        fetchData();
+
+        return () => {
+            is_cancelled = true;
```

Comment:
What is the purpose of this `is_cancelled` variable? It is used inside useEffect that is triggered on Component mount. And its value is set to **true** on mount lifecycle and to **false** on unmount.

Reply:
probably used for preventing memory leak console error. i have replaced it with useIsMounted. 👍🏼